File: backend/deafmute/src/main/java/dev/forint/deafmute/modules/recognitionconfig/entity/RecognitionConfig.java
package dev.forint.deafmute.modules.recognitionconfig.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("recognition_config")
public class RecognitionConfig {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String configName;

    private BigDecimal confidenceMin;

    private Integer holdMs;

    private Integer debounceMs;

    private Integer cooldownMs;

    private Integer requiredHits;

    private Integer maxIntervalMs;

    private Integer lockTimeoutMs;

    private Integer resetOnFail;

    private Integer allowRepeat;

    private String gestureOrderJson;

    private Integer activeFlag;

    private String remark;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}

File: adminend/v3-admin-vite-main/v3-admin-vite-main/src/common/apis/recognition-config.ts
import { request } from "@/http/axios"

export interface RecognitionConfigQuery {
  pageNum: number
  pageSize: number
  keyword?: string
  activeFlag?: number
}

export interface RecognitionConfigItem {
  id: number
  configName: string
  confidenceMin: number
  holdMs: number
  debounceMs: number
  cooldownMs: number
  requiredHits: number
  maxIntervalMs: number
  lockTimeoutMs: number
  resetOnFail: number
  allowRepeat: number
  gestureOrderJson: string | null
  activeFlag: number
  remark: string | null
  createTime: string
  updateTime: string
}

export interface RecognitionConfigFormData {
  configName: string
  confidenceMin: number
  holdMs: number
  debounceMs: number
  cooldownMs: number
  requiredHits: number
  maxIntervalMs: number
  lockTimeoutMs: number
  resetOnFail: number
  allowRepeat: number
  gestureOrderJson: string
  activeFlag: number
  remark: string
}

export interface PageResult<T> {
  list: T[]
  total: number
  pageNum: number
  pageSize: number
}

export interface ApiResponse<T> {
  code: number
  message: string
  data: T
}

export function getRecognitionConfigList(params: RecognitionConfigQuery) {
  return request<ApiResponse<PageResult<RecognitionConfigItem>>>({
    url: "/admin/recognition-config/list",
    method: "get",
    params
  })
}

export function getActiveRecognitionConfig() {
  return request<ApiResponse<RecognitionConfigItem>>({
    url: "/admin/recognition-config/active",
    method: "get"
  })
}

export function getRecognitionConfigDetail(id: number) {
  return request<ApiResponse<RecognitionConfigItem>>({
    url: `/admin/recognition-config/${id}`,
    method: "get"
  })
}

export function createRecognitionConfig(data: RecognitionConfigFormData) {
  return request<ApiResponse<null>>({
    url: "/admin/recognition-config",
    method: "post",
    data
  })
}

export function updateRecognitionConfig(id: number, data: RecognitionConfigFormData) {
  return request<ApiResponse<null>>({
    url: `/admin/recognition-config/${id}`,
    method: "put",
    data
  })
}

export function activateRecognitionConfig(id: number) {
  return request<ApiResponse<null>>({
    url: `/admin/recognition-config/${id}/activate`,
    method: "put"
  })
}
